lib/hawkular/prometheus/prometheus_api.rb

```diff
@@ -0,0 +1,50 @@
+require 'hawkular/base_client'
+require 'hawkular/inventory/entities'
+require 'ostruct'
+
+module Hawkular::Prometheus
+  class AlerterClient < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      entrypoint = normalize_entrypoint_url entrypoint, 'hawkular/alerter'
+      @entrypoint = entrypoint
+
+      super(entrypoint, credentials, options)
+    end
+
+    def prometheus_entrypoint
+      rest_client('/prometheus/endpoint').get
+    end
+  end
+
+  # Interface to talk with the Prometheus server used for Middleware Manager
+  # @param entrypoint [String] base url of Hawkular Services
+  class Client < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      prometheus_entrypoint = AlerterClient.new(entrypoint, credentials, options).prometheus_entrypoint
```

Comment:
Doesn't seem to me like the `Alerter` should be responsible to define the prometheus entrypoint. It should, imho, be defined on the client itself.

Reply:
From users perspective, P8S is given by MW Manager. MW Manager resolves the endpoint where P8S is hosted, so, this is just the flow to resolve this.